api/config_entry_routes.go

```diff
@@ -69,6 +72,20 @@ func (a *TCPRouteConfigEntry) GetModifyIndex() uint64 {
 	return a.ModifyIndex
 }
 
+// TCPService is a service reference for a TCPRoute
+type TCPService struct {
+	Name   string
+	Weight int
```

Comment:
Feels important to document the fields on this struct since these are the ones external API consumers will be interacting with I believe